packages/dev/core/src/AudioV2/abstractAudio/spatialAttachers/spatialAudioAttacher.ts

```diff
@@ -0,0 +1,205 @@
+
+export const _SpatialAudioAttachedEntity = {
+    CAMERA: "Camera",
+    MESH: "Mesh",
+    TRANSFORM_NODE: "TransformNode",
+};
+
+export const enum SpatialAudioAttachmentType {
+    POSITION = 1,
+    ROTATION = 2,
+    POSITION_AND_ROTATION = 3,
+}
+
+/**
+ * Provides a common interface for attaching an audio listener or source to a specific entity, ensuring only one entity
+ * is attached at a time.
+ */
+export class _SpatialAudioAttacher {
+    private _attachedEntity: Nullable<AbstractMesh | Camera | TransformNode> = null;
+    private _attacher: Nullable<_AbstractSpatialAudioAttacher> = null;
+    private _attachmentType: SpatialAudioAttachmentType;
+    private _isReadyPromise: Nullable<Promise<void>> = null;
+    private _minUpdateTime: number = 0;
+    private _spatialAudioNode: ISpatialAudioNode;
+
+    /**
+     * Creates a new ExclusiveSpatialAudioAttacher.
+     * @param spatialAudioNode - The spatial audio node to attach to
+     * @param attachmentType - The type of attachment to use; position, rotation or both. Defaults to both
+     */
+    public constructor(spatialAudioNode: ISpatialAudioNode, attachmentType: SpatialAudioAttachmentType = SpatialAudioAttachmentType.POSITION_AND_ROTATION) {
+        this._attachmentType = attachmentType;
+        this._spatialAudioNode = spatialAudioNode;
+    }
+
+    /**
+     * The scene that the audio listener or source is attached to, or null if the audio listener or source is not
+     * attached to a scene.
+     */
+    public get attachedCamera(): Nullable<Camera> {
+        return this._attacher?.getClassName() === _SpatialAudioAttachedEntity.CAMERA ? (this._attachedEntity as Camera) : null;
+    }
+
+    public set attachedCamera(value: Nullable<Camera>) {
+        if (this.attachedCamera === value) {
+            return;
+        }
+
+        this._isReadyPromise = this._resetAttachedEntity(value, _SpatialAudioAttachedEntity.CAMERA);
+    }
+
+    /**
+     * The mesh that the audio listener or source is attached to, or null if the audio listener or source is not
+     * attached to a mesh.
+     */
+    public get attachedMesh(): Nullable<AbstractMesh> {
+        return this._attacher?.getClassName() === _SpatialAudioAttachedEntity.MESH ? (this._attachedEntity as AbstractMesh) : null;
+    }
+
+    public set attachedMesh(value: Nullable<AbstractMesh>) {
+        if (this.attachedMesh === value) {
+            return;
+        }
+
+        this._isReadyPromise = this._resetAttachedEntity(value, _SpatialAudioAttachedEntity.MESH);
+    }
+
+    /**
+     * The transform node that the audio listener or source is attached to, or null if the audio listener or source is
+     * not attached to a transform node.
+     */
+    public get attachedTransformNode(): Nullable<TransformNode> {
+        return this._attacher?.getClassName() === _SpatialAudioAttachedEntity.TRANSFORM_NODE ? (this._attachedEntity as TransformNode) : null;
```

Comment:
as mesh inherits from transformNode, do we need a distinction? can we just support transformNode?

Reply:
The old audio engine centers the spatial audio in the mesh's bounding box, which is what `attachedMesh` does in the new audio engine. It will be useful when the old audio engine is reimplemented with the new audio engine.

Are you thinking it's not useful for users? Maybe it should be kept but marked as internal?